playbooks/phobos_vpn.yml

```diff
@@ -1,35 +1,111 @@
 
+    - name: Test connectivity to VPN server
+      # NOTE(coreywright): The Phobos VPN server cannot be pinged from within
+      # the Phobos environment, so use this task as a method of determining
+      # whether we are in the Phobos environment or not.
+      command: ping -c 3 {{ gateway }}
+      ignore_errors: true
+      changed_when: false
+      register: ping_vpn
 
-    - name: Template vpnc config
-      template:
-        src: vpnc.conf.j2
-        dest: /etc/vpnc/phobos.conf
+    - name: Configure and start VPN client
+      # NOTE(coreywright): If we are not in the Phobos environment, ie we can
+      # ping the Phobos VPN server, then we need to connect to it by way of
+      # VPN.
+      when: ping_vpn | succeeded
+      block:
+        # Use shell because python-apt isn't available so the apt module fails.
+        # I don't want to add python-apt to one of the more general slave setup
+        # scripts as it depends on python. We want to reduce the number of deps
+        # installed on general slaves to the bare minimum.
+        - name: Install apt packages
+          shell: |
+            apt-get update
+            apt-get install -y vpnc
+          register: install_packages
+          until: install_packages | success
+          retries: 5
+          delay: 2
+          tags:
+            - skip_ansible_lint
 
-    - name: Template vpnc systemd service
-      template:
-        src: vpnc.service.j2
-        dest: /lib/systemd/system/vpnc.service
 
-    - name: Start vpnc service
-      systemd:
-        enabled: yes
-        service: vpnc.service
-        daemon_reload: yes
-        state: restarted
+        - name: Template vpnc config
+          template:
+            src: vpnc.conf.j2
+            dest: /etc/vpnc/phobos.conf
+
+        - name: Template vpnc systemd service
+          template:
+            src: vpnc.service.j2
+            dest: /lib/systemd/system/vpnc.service
+
+        - name: Start VPN
+          block:
+            - name: Start vpnc service
+              systemd:
+                enabled: yes
+                service: vpnc.service
+                daemon_reload: yes
+                state: restarted
+              register: vpnc_restart
+
+          rescue:
+            - name: Output failed task information
+              debug: var=vpnc_restart
+
+            - name: Get vpnc service status
+              command: systemctl status vpnc.service
+              register: vpnc_status
+              failed_when: false
+              changed_when: false
+
+            - name: Output vpnc service status
+              debug: var=vpnc_status
+
+            - name: Get system log
+              command: journalctl -xe
+              register: system_log
+              changed_when: false
+
+            - name: Output system log
+              debug: var=system_log
+
+            - name: Exit with failure
+              fail: msg="vpnc service failed to start"
+
+    - name: Set MTU to 1450
```

Comment:
If I understand correctly, all phobos nodes need the MTU tweak, but not all jobs that may run in phobos include this playbook. 

So this should probably be in a separate playbook or a common.groovy function. 

eg:
```
void phobosSetMTU(){
  sh """
	which ping || { apt-get update; apt-get install -y iputils-ping}
	ping -c 3 ${gateway} || ip link set mtu 1450 dev ens3
  """
```
Then add a call to that after `install_ansible()` in [common.use_node](https://github.com/rcbops/rpc-gating/blob/master/pipeline_steps/common.groovy#L1008)

Is `ens3` the correct interface name across virtualised and bare metal instances?

Reply:
tl;dr this task is/was just a *temporary* work-around so i can/could continue testing and using this playbook until REO-365 was resolved.  sorry for not using the word "temporary", "short-term", etc and/or adding the "status-dont-merge" tag.

REO-365 has been resolved.  i was 99% sure of the resolution and the speed of its implementation, but deferred on doing it myself, so i implemented this work-around (and took great advantage of it in the intervening time).